ckanext/switzerland/logic.py

```diff
@@ -157,3 +163,106 @@ def highlight(term, q):
+    else:
+        log.error('Configuration missing for number of jobs to keep when clearing harvest sources')
+        raise ValidationError('Configuration missing for number of jobs to keep when clearing harvest sources')
+
+    log.info('Got config for number of most recent jobs per source to keep: {}'.format(number_of_jobs_to_keep))
+
+    # init list of jobs and objects to delete
+    delete_jobs_ids_all = []
+    delete_objects_ids_all = []
+
+    for source in sources_to_clear:
+        # get jobs ordered by their creations date
+        delete_jobs = model.Session.query(HarvestJob) \
+            .filter(HarvestJob.source_id == source) \
+            .order_by(HarvestJob.created.desc()).all()
+
+        # decide which jobs to keep or delete on their order
+        delete_jobs_ids = [job.id for job in delete_jobs][number_of_jobs_to_keep:]
+        keep_jobs_ids = [job.id for job in delete_jobs][:number_of_jobs_to_keep]
+
+        # log all job for a source with the decision to delete or keep them
+        log.info('Clear harvest source {}'.format(source))
+        log.info('- jobs_to_keep: {}'.format(','.join(keep_jobs_ids)))
+        for job in delete_jobs[:number_of_jobs_to_keep]:
+            log.debug('    - job {}: created:{}, status:{}'.format(job.id, job.created, job.status))
+        log.info('- jobs to delete: {}'.format(','.join(delete_jobs_ids)))
+        for job in delete_jobs[number_of_jobs_to_keep:]:
+            log.debug('    - job {}: created:{}, status:{}'.format(job.id, job.created, job.status))
+
+        # add to list of jobs to delete
+        delete_jobs_ids_all.extend(delete_jobs_ids)
+
+    # log all jobs to delete
+    log.info('Jobs to delete in total {}'.format(delete_jobs_ids_all))
+
+    if delete_jobs_ids_all:
+        # get harvest objects for harvest jobs
+        delete_objects = model.Session.query(HarvestObject.id) \
+           .filter(HarvestObject.harvest_job_id.in_(delete_jobs_ids_all)).all()
+        delete_objects_ids_all = list(itertools.chain(*delete_objects))
+
+        # perform delete
+        sql = '''begin;
+        delete from harvest_object_error where harvest_object_id in ('{delete_objects_values}');
+        delete from harvest_object_extra where harvest_object_id in ('{delete_objects_values}');
+        delete from harvest_object where id in ('{delete_objects_values}');
+        delete from harvest_gather_error where harvest_job_id in ('{delete_jobs_values}');
+        delete from harvest_job where id in ('{delete_jobs_values}');
+        commit;
+        '''.format(delete_objects_values="','".join(delete_objects_ids_all),
+                   delete_jobs_values="','".join(delete_jobs_ids_all))
+        model = context['model']
+        model.Session.execute(sql)
+
+        # lo delete
+        log.info('Clear {} harvest jobs'.format(len(delete_jobs_ids_all)))
+        log.info('Clear {} harvest objects'.format(len(delete_objects_ids_all)))
```

Comment:
In general we have a lot of logging going on here. Please make sure that the logging is meaningful. I have the feeling that quite a few log-entries are rather to debug.
We do not want to feed the ckan-logs unnecessarily when we are trying to get rid of parts of the harvester logs ;)

Reply:
I took quite some time to make the logging meaningful. But I agree to take it from info to debug.